apollo-router/src/configuration/mod.rs

```diff
@@ -525,75 +530,74 @@ impl Supergraph {
         path: Option<String>,
         introspection: Option<bool>,
         preview_defer_support: Option<bool>,
-        cache_redis_urls: Option<Vec<String>>,
+        apq: Option<Apq>,
+        query_planning: Option<QueryPlanning>,
     ) -> Self {
         Self {
             listen: listen.unwrap_or_else(test_listen),
             path: path.unwrap_or_else(default_graphql_path),
             introspection: introspection.unwrap_or_else(default_graphql_introspection),
             preview_defer_support: preview_defer_support.unwrap_or_else(default_defer_support),
-            cache_redis_urls,
+            apq: apq.unwrap_or_default(),
+            query_planning: query_planning.unwrap_or_default(),
         }
     }
 }
 
-#[cfg(not(feature = "experimental_cache"))]
-#[buildstructor::buildstructor]
-impl Supergraph {
-    #[builder]
-    pub(crate) fn new(
-        listen: Option<ListenAddr>,
-        path: Option<String>,
-        introspection: Option<bool>,
-        preview_defer_support: Option<bool>,
-    ) -> Self {
-        Self {
-            listen: listen.unwrap_or_else(default_graphql_listen),
-            path: path.unwrap_or_else(default_graphql_path),
-            introspection: introspection.unwrap_or_else(default_graphql_introspection),
-            preview_defer_support: preview_defer_support.unwrap_or_else(default_defer_support),
-        }
+impl Default for Supergraph {
+    fn default() -> Self {
+        Self::builder().build()
     }
 }
 
-#[cfg(not(feature = "experimental_cache"))]
-#[cfg(test)]
-#[buildstructor::buildstructor]
-impl Supergraph {
-    #[builder]
-    pub(crate) fn fake_new(
-        listen: Option<ListenAddr>,
-        path: Option<String>,
-        introspection: Option<bool>,
-        preview_defer_support: Option<bool>,
-    ) -> Self {
-        Self {
-            listen: listen.unwrap_or_else(test_listen),
-            path: path.unwrap_or_else(default_graphql_path),
-            introspection: introspection.unwrap_or_else(default_graphql_introspection),
-            preview_defer_support: preview_defer_support.unwrap_or_else(default_defer_support),
-        }
-    }
+#[derive(Debug, Clone, Default, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct Apq {
+    pub(crate) experimental_cache: Cache,
```

Comment:
```suggestion
    #[serde(rename = "experimental_cache")]
    pub(crate) cache: Cache,
```

It will be one line change if we stabilize it